src/lib/gssapi/krb5/k5sealv3.c

```diff
@@ -128,7 +130,14 @@ gss_krb5int_make_seal_token_v3 (krb5_context context,
             return err;
 
         /* Get size of ciphertext.  */
-        bufsize = 16 + krb5_encrypt_size (plain.length, key->keyblock.enctype);
+        encrypt_size = krb5_encrypt_size(plain.length, key->keyblock.enctype);
+        if ((encrypt_size == (size_t) - 1) ||
+            (encrypt_size > SIZE_MAX - 300)) {
```

Comment:
Also, (size_t)-1 is > SIZE_MAX - 300, so these clauses are redundant.

Using SIZE_MAX / 2 would be reasonable.  The caller will never have an object that large, and it would give us lots of margin to avoid overflows since we don't multiply here.